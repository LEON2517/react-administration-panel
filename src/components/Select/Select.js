/**
 * @modules components
 */
import React from 'react';
import PropTypes from 'prop-types';

import { createElement } from 'common';

export const HtmlSelect = createElement('select', 'Select', 'form-control');
export const Option = createElement('option', 'Option');

/**
 * компонент Селекс
 *
 * @param {Object} props  - props
 * @property {Array<string>} options - массив для option
 * @returns {ReactElememt}
 */
export default function Select(props) {
  const { options, ...rest } = props;
  return (
    <HtmlSelect {...rest}>
      { options.map(option => (<Option key={option} value={option} >{option}</Option>))}
    </HtmlSelect>
  );
}

Select.propTypes = {
  options: PropTypes.arrayOf(PropTypes.string),
};

Select.defaultProps = {
  options: [],
};
