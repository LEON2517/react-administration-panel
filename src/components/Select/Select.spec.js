/* @flow */
import React from 'react';
import { shallow } from 'enzyme';

import Select, { Option } from './Select';

describe('Select spec', () => {
  it('should render from string array', () => {
    const options = ['foo', 'bar'];
    const component = shallow(<Select options={options} />);
    expect(component.find(Option).length).toEqual(2);
    const firstOption = component.find(Option).first();
    expect(firstOption.props()).toEqual({
      value: 'foo',
      children: 'foo',
    });
    expect(component).toMatchSnapshot();
  });
});
