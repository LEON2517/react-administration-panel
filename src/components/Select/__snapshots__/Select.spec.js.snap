// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select spec should render from string array 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Select
    options={
      Array [
        "foo",
        "bar",
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <Option
          value="foo"
        >
          foo
        </Option>,
        <Option
          value="bar"
        >
          bar
        </Option>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "foo",
        "nodeType": "function",
        "props": Object {
          "children": "foo",
          "value": "foo",
        },
        "ref": null,
        "rendered": "foo",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "bar",
        "nodeType": "function",
        "props": Object {
          "children": "bar",
          "value": "bar",
        },
        "ref": null,
        "rendered": "bar",
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Option
            value="foo"
          >
            foo
          </Option>,
          <Option
            value="bar"
          >
            bar
          </Option>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "foo",
          "nodeType": "function",
          "props": Object {
            "children": "foo",
            "value": "foo",
          },
          "ref": null,
          "rendered": "foo",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "bar",
          "nodeType": "function",
          "props": Object {
            "children": "bar",
            "value": "bar",
          },
          "ref": null,
          "rendered": "bar",
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
